skcosmo/feature_selection/simple_fps.py

```diff
@@ -0,0 +1,129 @@
+import numpy as np
+import numbers
+
+from sklearn.utils.validation import NotFittedError
+from ._greedy import GreedySelector
+
+
+class SimpleFPS(GreedySelector):
+    """Transformer that performs Greedy Feature Selection using Farthest Point Sampling.
+
+
+    Parameters
+    ----------
+
+    n_features_to_select : int or float, default=None
+        The number of features to select. If `None`, half of the features are
+        selected. If integer, the parameter is the absolute number of features
+        to select. If float between 0 and 1, it is the fraction of features to
+        select.
+
+    initialize: int or 'random', default=0
+        Index of the first feature to be selected. If 'random', picks a random
+        value when fit starts.
+
+    progress_bar: boolean, default=False
+                  option to use `tqdm <https://tqdm.github.io/>`_
+                  progress bar to monitor selections
+
+    tolerance: float, defaults to 1E-12
+                threshold below which distances will be considered 0
+
+    Attributes
+    ----------
+    haussdorf_ : ndarray of shape (n_features,)
+                 the minimum distance from each feature to the set of selected
+                 features. once a feature is selected, the distance is not updated;
+                 the final list will reflect the distances when selected.
+    n_features_to_select : int
+        The number of features that were selected.
+
+    norms_ : ndarray of shape (n_features,)
+        The self-covariances of each of the features
+
+    X_selected_ : ndarray (n_samples, n_features_to_select)
+                  The features selected
+
+    selected_idx_ : ndarray of integers
+                    indices of the selected features, with respect to the
+                    original fitted matrix
+
+    support_ : ndarray of shape (n_features,), dtype=bool
+        The mask of selected features.
+
+    """
+
+    def __init__(
+        self,
+        n_features_to_select=None,
+        initialize=0,
+        tolerance=1e-12,
+        progress_bar=False,
+    ):
+
+        scoring = self.score
+        self.initialize = initialize
+
+        super().__init__(
+            scoring=scoring,
+            n_features_to_select=n_features_to_select,
+            progress_bar=progress_bar,
+            score_thresh_to_select=tolerance,
+        )
+
+    def _get_norms(self, X, y):
```

Comment:
I think a separate 1-line function is redundant. Especially since each additional function call technically increases the running time of the program slightly.

Reply:
This is to facilitate using PCov-variants in the future -- if we have a `get_norms` function, then we only need to overwrite the `get_norms`, `calculate_distances` and `_more_tags` functions. Is that too much emphasis on inheritability? Ideally `get_norms` should only be called once.

Reply:
This is an interesting question. In my humble opinion, if the function is only called once, there is no need to complicate the code unnecessarily. Although in general the idea is very nice, I understand your motives.

Reply:
I think it could go either way, tbh.

Reply:
Ok, it's up to you. Again, I understand why you want to add this feature, it just confuses me a bit.

Reply:
Okay! I'll revert and if it becomes an issue in the future I'll make a separate change

Reply:
Cool. Well, that's it, I think I've got everything I wanted :).